### PR TITLE
Add typings for `puppeteer` (headless chrome framework)

### PR DESCRIPTION
**Pull request by marvinhagemeister**
This PR adds typings for [puppeteer](https://github.com/GoogleChrome/puppeteer/), a framework by google around headless chrome.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

File: types/puppeteer/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+   * Whether to pipe browser process stdout and stderr into process.stdout and
+   * process.stderr. Defaults to false.
+   */
+  dumpio: boolean;
+  /** Path to a User Data Directory. */
+  userDataDir: string;
+}
+
+export function launch(options?: LaunchOptions): Promise<Browser>;
+export function myOtherMethod(a: number): number;
```
Comment (uniqueiniquity) [inline types/puppeteer/index.d.ts]: Did you mean to leave this in here?

**Review by uniqueiniquity — CHANGES_REQUESTED** [review]

File: types/puppeteer/index.d.ts (hunk 1 above)
Comment (marvinhagemeister) [reply types/puppeteer/index.d.ts]: Oh shoot, totally missed that!

Comment (marvinhagemeister) [reply types/puppeteer/index.d.ts]: @uniqueiniquity Fixed now. Thanks again for your review 👍

**Review by uniqueiniquity — APPROVED** [review]

**Comment by julien-c** [comment]
Should you ask the `puppeteer` maintainers if they'd be ok with sticking this into their repo (and maybe at some point autogenerate) @marvinhagemeister?

**Comment by marvinhagemeister** [comment]
@julien-c That would be ideal, but I must admit that I haven't heard of a tool that can convert standard `.js` files to `*.d.ts` reliably.

I've tried to for a handful of packages in the past. The author's response was usually one of:

- as long as this can't be automated, this puts too much burden on the author's that mostly don't care about TS
- if TS would be included, soon other languages like `flow` would want their types included as well, thus leading to more maintenance work

But these were just my experiences. Haven't asked the `puppeteer` guys specifically, though.

**Comment by julien-c** [comment]
@marvinhagemeister I think they are indeed working on it: https://github.com/GoogleChrome/puppeteer/issues/936

**Pull request merged**